### PR TITLE
nixos/tests: fix flaky cntr test

### PR DESCRIPTION
**Pull request by JJJollyjim**
###### Motivation for this change

The cntr test sometimes hangs until the 10-hour hydra limit – this is currently the sole thing stopping trunk-combined from advancing. 

###### Things done

This behaviour appears to be an edge-case related to the type of TTY in which the cntr command runs during test execution. We can work around this by running the command as a background job.

I additionally added a wait_for_open_port to fix nondeterministic test failures I observed after fixing the hanging issue.

!ofborg test cntr

### CODE REVIEW

**Comment by JJJollyjim** [comment]
Force-pushed since I left some debugging tools in

!ofborg test cntr

**Comment by JJJollyjim** [comment]
… and again with the trailing whitespace removed :)

!ofborg test cntr

**Comment by zowoq** [comment]
> this is currently the sole thing stopping trunk-combined from advancing.

`cntr` isn't a channel blocker?

**Comment by zowoq** [comment]
Needs to be `@` not `!` .

@ofborg test cntr

**Comment by JJJollyjim** [comment]
> `cntr` test isn't a channel blocker?

Ah I guess not by the standard definition. It "blocks" the channel from advancing for 10 hours while we wait for it to time out, but it no, it's not a blocker blocker.

**Comment by zowoq** [comment]
`ofborg` doesn't seem to like `cntr`

@ofborg test cntr.docker cntr.nixos-container cntr.podman

**Review by rhysmdnz — APPROVED** [review]

File: nixos/tests/cntr.nix [hunk 1]
```diff
@@ -28,10 +28,16 @@ let
       testScript = ''
         start_all()
         ${backend}.wait_for_unit("${backend}-nginx.service")
-        result = ${backend}.wait_until_succeeds(
-            "cntr attach -t ${backend} nginx sh -- -c 'curl localhost | grep Hello'"
+        ${backend}.wait_for_open_port(8181)
+        # For some reason, the cntr command hangs when run without the &.
+        # As such, we have to do some messy things to ensure we check the exitcode and output in a race-condition-safe manner
```
Comment (Mic92) [inline nixos/tests/cntr.nix:33]: This seems weird. Maybe something with the terminal changed in our nixos tests?
Can you also open up an issue in cntr itself? Should be fine to merge this though in the meantime.

**Pull request merged**